src/passes/Asyncify.cpp

```diff
@@ -795,30 +802,35 @@ static bool doesCall(Expression* curr) {
 class AsyncifyBuilder : public Builder {
 public:
   Module& wasm;
+  bool is64;
+  Type pointerType;
 
-  AsyncifyBuilder(Module& wasm) : Builder(wasm), wasm(wasm) {}
+  AsyncifyBuilder(Module& wasm)
+    : Builder(wasm), wasm(wasm), is64(wasm.memories[0]->is64()),
+      pointerType(is64 ? Type::i64 : Type::i32) {}
 
   Expression* makeGetStackPos() {
     return makeLoad(4,
                     false,
                     int32_t(DataOffset::BStackPos),
                     4,
-                    makeGlobalGet(ASYNCIFY_DATA, Type::i32),
-                    Type::i32,
+                    makeGlobalGet(ASYNCIFY_DATA, pointerType),
+                    pointerType,
                     wasm.memories[0]->name);
   }
 
   Expression* makeIncStackPos(int32_t by) {
     if (by == 0) {
       return makeNop();
     }
+    Literal literal = is64 ? Literal(int64_t{by}) : Literal(int32_t{by});
```

Comment:
```suggestion
    auto literal = Literal::makeFromInt64(by, pointerType);
```
I think with this we can avoid the `bool is64` on the class.

Reply:
(I mean with changes like this and the one below - then all we need is `pointerType`)